src/Login/Signup.js

```diff
@@ -29,15 +33,18 @@ const SignUp = () => {
         onChangeText={setEmail}
         keyboardType="email-address"
         autoCapitalize="none"
-      />
+        />
       <TextInput
         style={styles.input}
         placeholder="Password"
         value={password}
         onChangeText={setPassword}
         secureTextEntry
-      />
-      <Button title="Sign Up" onPress={handleSignUp} />
+        />
+
+        <TouchableOpacity style={[styles.button, !email || !password ? styles.disabledButton : null]} disabled={!email || !password} onPress={handleSignUp}>
```

Comment:
!email || !password can be store in a variable, so that the disabled property and style condition can be checked with single variable which makes it a bit clear code